libs/usb-drive/src/usb_drive.ts

```diff
@@ -319,8 +319,8 @@ export function detectUsbDrive(logger: Logger): UsbDrive {
       }
       await logEjectInit(logger, loggingUserRole);
       try {
-        await unmountUsbDrive(deviceInfo.mountpoint);
         didEject = true;
+        await unmountUsbDrive(deviceInfo.mountpoint);
```

Comment:
@jonahkagan while testing the previous commit, I actually hit another issue I've never hit before. In a dev environment with a real USB drive, my USB drive would sometimes remount immediately after ejecting. In other words, it didn't seem like it was respecting the `ejected` state. Looking at the logging, it looked like the mount was getting triggered via `status` before `umountUsbDrive` exited. By moving the `didEject=true` status a bit earlier, I'm not longer able to reproduce. 

It means it's possible that `didEject` is set to `true` and the unmount actually fails. But if we're in a situation where the unmount fails, we probably have other problems, and there's not much we can do about. And removing the drive will reset the status.

Reply:
Sounds like there's a race condition between mounting and unmounting.

I think optimistically setting `didEject = true` is one way to handle it, but you're right it's a bit messy.

A safter approach might be to have a lock that the mounting/unmounting actions share to avoid contention. This could be as simple as an `isUnmounting` flag that we check before trying to mount.

Reply:
Now that I'm looking into this more closely, we have a few different "busy" states that we want to lock on (mounting, formatting, unmounting). It might be worth thinking about how to easily prevent actions if the drive is busy in any way. E.g. we could have a single `isBusy` flag. Or have multiple flags an a helper that checks if any of them is set.